anycloud/node/src/datastore.ts

```diff
@@ -1,32 +1,67 @@
-import { dsdel, dsgetv, dshas, dssetv } from 'alan-js-runtime';
-
-const ns = 'kv';
-
-export const DS = new Proxy({}, {
-  get: function (_, dsKey) {
-    const dsValue = dsgetv(ns, dsKey);
-    return dsValue.length === 2 ? dsValue[1] : 'Value not found';
-  },
-  set: function (_, dsKey, dsValue) {
-    if (dshas(ns, dsKey)) { return false; }
-    dssetv(ns, dsKey, dsValue)
-    return true;
-  },
-  deleteProperty: function (_, dsKey) {
-    if (!(dshas(ns, dsKey))) { return false; }
-    return dsdel(ns, dsKey);
-  },
-  has: function (_, dsKey) {
-    return dshas(ns, dsKey);
-  },
-});
-
-
-console.log(DS['foo'] = 'bar');
-console.log(DS['foo']);
-console.log('foo' in DS);
-console.log(delete DS['foo']);
-console.log('foo' in DS);
-console.log(DS['foo'] = {foo1: "bar1"});
-console.log(DS['foo']);
-console.log(delete DS['foo']);
+require('cross-fetch/polyfill')
+
+class DataStore {
+  private localDS: any;
+  private ns: string;
+  private clusterSecret?: string;
+  private isLocal: boolean;
+  private headers?: any;
+  private ctrlPortUrl: string;
+
+  constructor() {
+    this.localDS = {};
+    this.ns = 'kv';
+    this.clusterSecret = process.env.CLUSTER_SECRET;
+    this.isLocal = this.clusterSecret ? false : true;
+    this.headers = this.clusterSecret ? {[this.clusterSecret]: 'true'} : null;
+    this.ctrlPortUrl = 'https://localhost:4142';
+  }
+
+  private async request(method: string, url: string, body?: any): Promise<Response> {
+    console.log('localpath', this.isLocal);
+    return fetch(url, {
+      method: method,
+      headers: this.headers,
+      body,
+    });
+  }
```

Comment:
I would `await` on the `fetch` call in here and return the promise for the body.

Otherwise:
1. This doesn't need to be an `async` function at all
2. You're requiring users of the functions below to `await (await ds.get('foo'))` which is awkward.

Reply:
I've change the return values of all functions and did some changes before reading this comment. But I'm still returning the fetch. I guess a could `await` here as you suggest and just return the response to be handled by each method.